compiler/grainformat/reformat.re

```diff
@@ -2068,87 +2045,90 @@ and print_other_application =
     print_infix_application(~expressions, ~original_source, ~comments, func)
   | _ when infixop(function_name) =>
     raise(Error(Illegal_parse("Formatter error, wrong number of args ")))
-  | _ =>
-    if (function_name == list_cons) {
-      resugar_list(~original_source, ~comments, expressions);
-    } else if (Array.exists(fn => function_name == fn, exception_primitives)) {
-      let first_expr = List.hd(expressions);
-      Doc.concat([
-        print_expression(~original_source, ~comments, func),
-        Doc.space,
-        print_expression(~original_source, ~comments, first_expr),
-      ]);
-    } else {
-      // standard function application
-      // look out for special cases of callbacks in first or last position
+  | _ when function_name == list_cons =>
+    resugar_list(~original_source, ~comments, expressions)
+  | [first_expr, ..._]
```

Comment:
Be careful with your ordering of pattern matches, since they match top-to-bottom, I'm always concerned when there's a `_` above the other matches. In this case it has a specific `when` clause, so it should only match in one case, but again, it scares me when I see it.

Reply:
(I like the change to `when` clauses though!)